### PR TITLE
Asynchronous read partitions

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Asynchronous read partitions

## Changelog

- Improvement

## Related Issues

#2492

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
Wow, I'm convinced @leiysky interested in this PR :rocket:

File: query/src/sql/sql_parser.rs [hunk 1]
```diff
@@ -653,6 +706,7 @@ impl<'a> DfParser<'a> {
     }
 
     fn parse_truncate(&mut self) -> Result<DfStatement, ParserError> {
+        self.parser.next_token();
         match self.parser.next_token() {
```
Comment (sundy-li) [inline query/src/sql/sql_parser.rs:710]: So many `match self.parser.next_token()`

Let's refactor this into `self.parser.expect_token(token)?`

Comment (zhang2014) [reply query/src/sql/sql_parser.rs:728]: It has been verified `parse_statement match`,  may not be necessary to use `expect_token` ？

**Comment by sundy-li** [comment]
/LGTM

Since it passed all tests, let's merge it to avoid conflicts.

**Comment by BohuTANG** [comment]
/lgtm

Cool, Let's merge

**Comment by zhang2014** [comment]
related #1830

**Pull request merged**